From 05a5a585bcac9ab40729372515ca106c8d58aa61 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 7 May 2024 22:44:25 +0000
Subject: [PATCH] Dont abort if the DnsHelper failed to init on BPF-less kernel

---
 DnsProxyListener.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DnsProxyListener.cpp b/DnsProxyListener.cpp
index 302cfd99..515d3b55 100644
--- a/DnsProxyListener.cpp
+++ b/DnsProxyListener.cpp
@@ -693,7 +693,7 @@ IsUidBlockedFn resolveIsUidNetworkingBlockedFn() {
         // On S/Sv2 this can fail if tethering apex is too old, ignore it.
         if (ret == -EOPNOTSUPP && !isAtLeastT()) return nullptr;
         LOG(ERROR) << __func__ << ": ADnsHelper_init failed " << strerror(-ret);
-        abort();
+        return nullptr;
     }
 
     // Related BPF maps were only mainlined from T.
-- 
2.43.0

