From d79f4433506dfa0d3a12dd9250894a5cc33fd412 Mon Sep 17 00:00:00 2001
From: ethan <ethangoh98@gmail.com>
Date: Sat, 3 October 2026 1:41:32 +1100
Subject: [PATCH] removing NexusLauncher as it was causing crashes.

diff --git a/common/common-vendor.mk b/common/common-vendor.mk
index 6f35065..5f462e5 100644
--- a/common/common-vendor.mk
+++ b/common/common-vendor.mk
@@ -51,8 +51,7 @@ PRODUCT_PACKAGES += \
     WellbeingPrebuilt \
     GoogleFeedback \
     GoogleServicesFramework \
-    WallpaperPickerGoogleRelease \
-    NexusLauncherRelease
+    WallpaperPickerGoogleRelease
 
 PRODUCT_PACKAGES += \
     privapp_permissions_google \
diff --git a/common/proprietary/system_ext/priv-app/Android.bp b/common/proprietary/system_ext/priv-app/Android.bp
index d7ca860..443d3e0 100644
--- a/common/proprietary/system_ext/priv-app/Android.bp
+++ b/common/proprietary/system_ext/priv-app/Android.bp
@@ -37,17 +37,3 @@ android_app_import {
     privileged: true,
     system_ext_specific: true,
 }
-
-android_app_import {
-    name: "NexusLauncherRelease",
-    overrides: ["Launcher3", "Launcher3QuickStep"],
-    owner: "google",
-    apk: "NexusLauncherRelease/NexusLauncherRelease.apk",
-    presigned: true,
-    preprocessed: true,
-    dex_preopt: {
-        enabled: false,
-    },
-    privileged: true,
-    system_ext_specific: true,
-}
