From 64dcb393d072cf2a9c1467a9262aa333d78f5f64 Mon Sep 17 00:00:00 2001
From: Archfx <cybx@live.com>
Date: Sun, 14 Jul 2024 13:09:38 -0400
Subject: [PATCH] two seperate variants for duo1 and duo2

---
 generate.sh | 18 +++++++++++++++---
 2 files changed, 15 insertions(+), 3 deletions(-)


diff --git a/generate.sh b/generate.sh
index 4c7690a..8ce4d91 100644
--- a/generate.sh
+++ b/generate.sh
@@ -18,12 +18,13 @@ for part in a ab;do
 	for apps in vanilla gapps foss gapps-go;do
 		for arch in arm64 arm a64;do
 			for su in yes no;do
+			for duo in "" _duo1 _duo2; do
 				apps_suffix=""
 				apps_script=""
 				apps_name=""
 				extra_packages=""
                 vndk="vndk.mk"
-		optional_base=""
+				optional_base=""
 				if [ "$apps" == "gapps" ];then
 					apps_suffix="g"
 					[[ "$arch" == "a64" ]] && base="arm" || base="arm64"
@@ -57,7 +58,7 @@ for part in a ab;do
 					su_suffix='S'
 					extra_packages+=' phh-su me.phh.superuser'
 				fi
-
+				
 				part_suffix='a'
 				if [ "$part" == 'ab' ];then
 					part_suffix='b'
@@ -65,7 +66,16 @@ for part in a ab;do
 					optional_base='$(call inherit-product, device/phh/treble/base-sas.mk)'
 				fi
 
-				target="treble_${arch}_${part_suffix}${apps_suffix}${su_suffix}"
+				duo_suffix=''
+				if [ "$duo" == "_duo1" ];then
+					duo_suffix='_duo1'
+					extra_packages+=' PostureProcessor '
+				elif [ "$duo" == "_duo2" ];then
+					duo_suffix='_duo2'
+					extra_packages+=' PostureProcessorV2 '
+				fi
+
+				target="treble_${arch}_${part_suffix}${apps_suffix}${su_suffix}${duo_suffix}"
 
 				baseArch="$arch"
 				if [ "$arch" = "a64" ];then
@@ -98,6 +108,8 @@ PRODUCT_PACKAGES += $extra_packages
 
 EOF
 echo -e '\t$(LOCAL_DIR)/'$target.mk '\' >> AndroidProducts.mk
+
+			done
 			done
 		done
 	done
-- 
2.34.1

