From d79f4433506dfa0d3a12dd9250894a5cc33fd412 Mon Sep 17 00:00:00 2001
From: ethan <ethangoh98@gmail.com>
Date: Thu, 7 October 2026 5:21:32 +1100
Subject: [PATCH] Setup Gapps to use OpenGapps instead of ponces vendor gapps, uninstall v2 camera for SD2

diff --git a/device/phh/treble/aosp.mk b/device/phh/treble/aosp.mk
index f3a8fae..d21dc93 100644
--- a/device/phh/treble/aosp.mk
+++ b/device/phh/treble/aosp.mk
@@ -1,4 +1,4 @@
 $(call inherit-product, vendor/ponces/config/common.mk)
 
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    ro.system.ota.json_url=https://raw.githubusercontent.com/Archfx/duo-de/main-16/config/ota.json
\ No newline at end of file
+    ro.system.ota.json_url=https://raw.githubusercontent.com/Archfx/duo-de/main-16/config/ota.json
diff --git a/device/phh/treble/generate.sh b/device/phh/treble/generate.sh
old mode 100644
new mode 100755
index 50912a8..16d315e
--- a/device/phh/treble/generate.sh
+++ b/device/phh/treble/generate.sh
@@ -14,6 +14,21 @@ fi
 
 echo 'PRODUCT_MAKEFILES := \' > AndroidProducts.mk
 
+GAPPS_PRODUCT_PACKAGES="GAPPS_PRODUCT_PACKAGES += \\
+	CalculatorGoogle \\
+	CalendarGoogle \\
+	CameraGoogle \\
+	Chrome \\
+	ClockGoogle \\
+	ContactsGoogle \\
+	Gmail \\
+	Maps \\
+	Messenger \\
+	Photos \\
+	AndroidAuto \\
+	KeyboardGoogle"
+GAPPS_VARIANT="GAPPS_VARIANT := nano"
+
 for part in a ab;do
 	for apps in vanilla gapps foss gapps-go old;do
 		for arch in arm64 arm a64;do
@@ -21,13 +36,17 @@ for part in a ab;do
 				apps_suffix=""
 				apps_script=""
 				apps_name=""
+				gapps_variant=""
+				gapps_product_packages=""
 				extra_packages=""
                 vndk="vndk.mk"
 		optional_base=""
 				if [ "$apps" == "gapps" ];then
 					apps_suffix="g"
-					apps_script='$(call inherit-product, vendor/gapps/common/common-vendor.mk)'
+					apps_script='$(call inherit-product, vendor/opengapps/build/opengapps-packages.mk)'
 					apps_name="with GApps"
+					gapps_variant=$GAPPS_VARIANT
+					gapps_product_packages=$GAPPS_PRODUCT_PACKAGES
 				fi
 				if [ "$apps" == "gapps-go" ];then
 					apps_suffix="o"
@@ -101,6 +120,9 @@ PRODUCT_CHARACTERISTICS := device
 
 PRODUCT_PACKAGES += $extra_packages
 
+$gapps_variant
+$gapps_product_packages
+
 EOF
 echo -e '\t$(LOCAL_DIR)/'$target.mk '\' >> AndroidProducts.mk
 			done
diff --git a/device/phh/treble/phh-on-data.sh b/device/phh/treble/phh-on-data.sh
index a77f66f..e2ecea4 100644
--- a/device/phh/treble/phh-on-data.sh
+++ b/device/phh/treble/phh-on-data.sh
@@ -47,7 +47,7 @@ fi
 # Retrieve the hardware property
 hardware=$(getprop ro.hardware)
 if [ "$hardware" = "surfaceduo2" ]; then
-    pm uninstall -k --user 0 com.microsoft.surface.camera
+    pm uninstall -k --user 0 com.microsoft.surface.camera.v2
     settings put global enable_freeform_support 1
     echo "1" >> /sys/devices/platform/soc/soc:surface_util/ms_pen_charger/ms_pen_charger
     chmod 777 /sys/devices/platform/soc/soc:surface_util/ms_pen_charger/ms_pen_charger
